### ISSUE TITLE
Idea: Event Editor

### ISSUE THREAD

**Issue by r4j4h** [PropaneDragon/RushHour#19]
Rush Hour now features user customizable events. 

We can have them come from assets and I believe I saw @PropaneDragon encouraging users to make their own custom ones by creating .XML files manually. This inspired the idea in me for Rush Hour to provide an editor for these events to make it easier for users.

At the very least being able to view a list of the currently loaded XML files and the currently loaded events would be a big boon to let the user know what events are in play. Even stronger would be a way to enable/disable them. And then topping that would be a built in way to modify and add new ones.

I think this would increase and enable the userbase, which might not be as comfortable with files, file systems, and not to mention XML, to help generate events and to share events.

I'm not sure anyone's interest in this but if there is some this would definitely require some design ahead of time to do properly so toward that end I've attempted to start a design discussion below beginning with where the menu should live:
##### Tools Menu

The Tools Menu provides Map, Asset, and Theme editors and I feel this would be a great place for an Event Editor.

The menu shows buttons for those editors and manages showing/hiding little panels of New/Load buttons when any are clicked. Those buttons then generally open a Dialog for progress with the user. New/Load would work well a singular Event but in our context we are managing "event packages", if you will, of XML files. So New/Manage might work better. Then New might let you choose a name and then add/manage events within it and then save to an .XML file, while Manage lets you open an .XML file.

I did some preliminary feasibility research. While I am unfamiliar with that menu's code, most of it was simple to follow and understand. I lost tracing the graphics for the menu items or the actual "template" for the menus. My trail stopped at `...\Steam\steamapps\common\Cities_Skylines\Cities_Data\level4`. I don't know how to read those guys. My concern with understanding this is to ensure that the screen will be able to handle an additional button - e.g. will it be a scrollable panel, etc.

Not meaning to spend too much time on one avenue though, so let's move on.
##### Content Manager Menu as Events Category

The Content Manager has an extensible Categories list and then lists items for each Category. Existing Categories already provide precedents for configuration (events within an event package as opposed to one event per slot), for sharing and/or viewing on workshop, and for adding new entries of (See Add style from Styles category). We could introduce a new Events category.

The Mods category provides Options buttons like we would probably want to use, but with a label like Modify/Change/Edit/Manage/Customize.

On the precedent of Sharing and Workshop links, anything useful would probably also require cooperation of other parties like steam, but we could perhaps facilitate users in opening an Explorer window or Finder window or otherwise getting the path to their XML for manual sharing. We could also link to the source prop's Steam Workshop page if the file is coming from an asset.

Adding new events would facilitate users in using our provided GUI to easily and safely build events.
##### Rush Hour Options Sub-Menu

Rush Hour's own Options menu would make a fine home for viewing/managing events. Perhaps as a sub-menu within our options panel or a Dialog. I already desire organizing our Options panel better and this route could harmoniously work towards that. It also keeps the event management explicitly encapsulated by Rush Hour and stays under what's already in our domain of modification.

**Comment by PropaneDragon** [comment]
Having something in-game to create events would be really nice. I like the idea of having it available in the options, and I've already re-organised the options into tabs but I'm yet to push any changes up regarding this. When I've got things organised properly maybe we could make a start on this?

I need to decide what I'm doing with allowing players to create their own events. I think this needs planning out first before getting too into the random events.

**Comment by r4j4h** [comment]
I agree, I primarily wanted to open up a discussion around it with you and get my energy out into something tangible we can benefit from down the road. :) I will try to keep an eye out, but if I don't catch wind don't hesitate to let me know when you want to move forward on this sort of thing. Can't wait to see the new tabs!

**Comment by PropaneDragon** [comment]
I'll make sure to let you know. Also, the new options are up. Hopefully this is a bit tidier.

**Comment by PropaneDragon** [comment]
Don't know whether you're still working with Cities anymore, but it'd be nice to get this in at some point now events are sort of stable. Let me know if you want to work on this, or not.

**Comment by r4j4h** [comment]
Ah, I'm sorry for letting this go unresponded to for so long! I do intend on getting back to Cities and RushHour, but it will be a few weeks from now before I get the time, so feel free to get started without me if you want!

**Comment by PropaneDragon** [comment]
It's fine :). I ended up adding one in as an external program, as doing heavy UI stuff in Cities is just awful.
